Ensure Python 2.7 compatibility for UserData methods

alecpm opened the pull request:
This PR includes a small tweak to prevent the `session_mutator` wrapped `UserDict` methods from failing with `TypeError: unbound method __setitem__() must be called with UserDict instance as first argument (got BeakerSessionDataObject instance instead)`. If this seems too hacky, we could instead have `BeakerSessionDataObject` inherit from `UserDict` (which would allow removing the custom `get` and `__getitem__` methods).

alecpm committed:
Ensure Python 2.7 compatibility for UserData methods.

alecpm committed:
Add some additional dict compatibilty methods (without `__contains__` `in` is deadlocking for some reason).

davisagli reviewed (commented):
Looks okay to me, but please add a changelog entry before merging.

pgrunewald committed:
Add changelog entry for Python 2.7 compatibility

pgrunewald commented:
I took the liberty and added the missing changelog entry.

pgrunewald committed:
Use proper import for InitializeClass

The previous version via App.class_init.InitializeClass was already an alias and got removed in the meantime.

veritazx commented:
So we are back in compatibility land and also compatible with Zope 5.9 on Py3. Nice.
I'm going to merge it.

Thanks!
